[Shawn Ng Yong Kwang] iP

shawn-nyk opened the pull request.

Commit:
Add Gradle support

shawn-nyk committed:
Add greet, echo and exit functionality

Duke now greets the user, will echo the user's input, and the program
will exit when the user inputs "bye". Duke's name has also been
changed to "Star Bot", as in that is the name that it goes by during
user interaction, but the class and file name is left as "Duke" for the
purposes of this assignment.

shawn-nyk committed:
Add task list functionality

Duke can now keep a record of the user's inputs as a list of tasks and
can show the list to the user upon request.

shawn-nyk committed:
Create Task.java and add mark-as-done feature

Task.java encapsulates a task in the Duke program. Tasks now include an
"isDone" property that marks whether the task has been completed or
not, and users can now mark tasks as completed.

shawn-nyk committed:
Add ToDo, Event and Deadline Tasks

Duke now supports 3 types of tasks - ToDo, Event and Deadline tasks.

shawn-nyk committed:
Do automated text UI testing

Run an automated text UI test using the I/O redirection technique to
semi-automate the testing of all of Duke's features up til now.

shawn-nyk committed:
Handle errors

Create a number of exception classes specific to the Duke program to
handle different forms of invalid user input.

shawn-nyk committed:
Add deletion functionality

Users can now delete tasks from the task list. Also add error handling
to handle invalid user input regarding this delete command.

shawn-nyk committed:
Add save functionality

Duke is not able to save task lists upon exiting the program.

Users thus cannot keep a saved task list unless they do not
terminate the program.

Add save functionality in which Duke writes the task list to a file on
the hard disk whenever the task list changes. Also handle cases when
the path to the save file does not exist - save file or directories in
path to save file does not exist.

Saving to the hard disk this way follows the Level 7 requirements.

shawn-nyk committed:
Add dates and times functionality for deadlines

Deadline tasks do not specify that the user must enter a date and time.

This allows users to potentially give deadlines that are not meaningful
(that are not dates and times, e.g. "do task /by Mark" instead of "do
task /by tomorrow morning).

Deadline tasks should enforce that users have to specify a date and
time for the deadline. Let's use the java.time.LocalDate and
java.time.LocalDateTime classes to store the date and time specified by
the user. The resulting deadline command format is thus: "deadline task
/by YYYY-MM-DD hhmm" where hh is hours and mm is minutes.

The LocalDate and LocalDateTime classes are used for ease of storing
and manipulating the date and time data. In turn, the format for the
deadline command is used for ease of creating the LocalDate and
LocalDateTime objects.

shawn-nyk committed:
Merge branch 'branch-Level-7'

Add save functionality to master branch version of Duke.

shawn-nyk committed:
Merge branch 'branch-Level-8'

Add dates and times functionality to deadline tasks to master branch
version of Duke.

shawn-nyk committed:
Update automated text testing

Automated text testing can now be run repeatedly and takes into account
and tests the new changes made up til Level-8.

shawn-nyk committed:
Refactor code in Duke.java

Duke.java is handling many different responsibilities - including user
interaction, loading and saving files, and parsing and executing
commands.

This is not in line with the single responsibility principle that
states that a class should only manage a single part of a program's
functionality.

Let's refactor the code in Duke.java to:
* Ui.java to deal with interactions with the user
* Storage.java to deal with loading tasks from and saving tasks to the
save file
* Parser.java to deal with making sense of user commands
* TaskList.java to deal with manipulating the task list
* Command.java and its subclasses to define and execute user commands

This follows the single responsibility principle by creating classes
as necessary that each manage a single part of the program's
functionality.

shawn-nyk committed:
Organize classes into packages

All the java files are in one single directory.

This makes navigating the directory to access certain classes difficult
as there are many files to sieve through. It also does not provide
encapsulation that packages provide and does not make it clear upon
looking through the directory what types of classes there are - classes
that deal with user commands, classes that deal with tasks, and classes
that deal with excpetions.

Let's organize the classes into suitable java pacakages:
* duke - as the highest level package containing all other packages /
  java files
* duke.command - containing all classes that deal with user commands
* duke.exception - containing all duke excpetion classes
* duke.task - containing all classes that deal with tasks

This makes the code base neater and hence easier to navigate and
understand, and also allows for encapsulation via packages.

shawn-nyk committed:
Add JUnit tests

Add JUnit tests for the following classes:
* Parser
* Task
* ToDo
* Event
* Deadline

shawn-nyk committed:
Create JAR file

shawn-nyk committed:
Add JavaDocs to all files in src/main/java

shawn-nyk committed:
Tweak code to comply with coding standard

shawn-nyk committed:
Add task finding functionality

Duke cannot find tasks in the task list based on keywords.

This makes it inconvenient for users who want to do so, which is a
common feature for a task list.

Let's add task finding functionality in which users can specify
keywords and Duke will then search through the task list and return
tasks that contain those keywords, if any.

shawn-nyk committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDocs to all files in src/main/java

shawn-nyk committed:
Merge branch 'branch-A-CodingStandard'

Tweak code to comply with coding standard.

shawn-nyk committed:
Merge branch 'branch-Level-9'

Add task finding functionality to master branch version of Duke.

shawn-nyk committed:
Merge remote branch 'add-gradle-support'

Add Gradle support to Duke.

shawn-nyk committed:
Add Gradle support

Add and set up gradle to build and run Duke, as well as run unit tests.
Checkstyle yet to be done.

shawn-nyk committed:
Merge branch 'branch-A-Gradle'

Add Gradle support to master branch version of Duke.

shawn-nyk committed:
Implement GUI

Implement GUI for Duke.

shawn-nyk committed:
Merge branch 'branch-Level-10'

Add GUI to master branch version of Duke.

chloelee767 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,38 @@
+package duke;
+
+import duke.command.*;
```
Would it be better to list the imported classes separately?

chloelee767 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,115 @@
+
+    /**
+     * Writes to save file as specified by filePath. Writing to save occurs every time the task list changes.
+     *
```
Would it be better to include the function parameter in the JavaDoc?

chloelee767 commented on src/main/java/duke/Response.java:
```diff
@@ -0,0 +1,21 @@
+
+import duke.command.Command;
+
+public class Response {
```
Should this class be documented?

chloelee767 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,65 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
```
Would it be better to document this method (as well as getDukeDialog below)?

chloelee767 reviewed (commented):
Overall looks good, only a few minor coding standard violations.

foongsq commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,115 @@
+     * @throws FileNotFoundException If from the path specified by filePath, a save file is not found or there is a
+     * missing directory / there are missing directories.
+     */
+    public List<String> load() throws FileNotFoundException {
```
Perhaps parts of this function could be abstracted out into other functions to make the code more readable?

foongsq commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,215 @@
+
+public class ParserTest {
+    @Test
+    public void parse_validByeCommand_exitCommand() {
```
I like how you made the test names very descriptive.

foongsq reviewed (commented):
Overall, good job on giving all your variables and functions appropriate names. Other than 1 function that is longer than the optimal length of a function, I didn't find any other issues with your code. Keep up the good work!

shawn-nyk committed:
Comply with checkstyle rules

Edit all code (both main and test) to comply with checkstyle rules.

shawn-nyk committed:
Merge branch 'branch-A-CheckStyle'

Make code in master branch comply with checkstyle rules.

shawn-nyk committed:
Fix CLI cross and tick not appearing correctly

Duke's CLI version does not display the crosses and ticks of tasks
correctly ever since the GUI version was implemented.

This makes the CLI version not suitable for use.

To continue to support the CLI version of Duke, separate methods for
printing the correct crosses and ticks have been added.

shawn-nyk committed:
Generate new jar file

Generate new jar file for Duke v0.2 that now uses a GUI.

shawn-nyk committed:
Add assertions to main code

Document important assumptions that should hold at various points in
the program.

shawn-nyk committed:
Refactor code in duke package

Refactor code in duke package (not its subpackages) to improve code
quality.

shawn-nyk committed:
Refactor code in Parser and 2 related packages

Some command parsing is being done by command objects when they are
executed.

Parsing should be done solely by the Parser class (Single
Responsibility Principle).

Refactor code in Parser, and the command and task packages. Shift all
command parsing duties into Parser. Adhere to CS2103T code quality
standards as well.

zhaolingshan reviewed (commented):
I like how your code is very easy to read and understand! Java docs is done very well but missing for a few methods, and I especially like how you name the methods very clearly. Your code follows the java coding standard completely. great work!

shawn-nyk committed:
Refactor code in Parser and related packages more

Further improve code quality in Parser and the command and task
packages. Minor refactoring also done in Duke and TaskList.

shawn-nyk committed:
Refactor code in exception package

shawn-nyk committed:
Merge pull request #1 from shawn-nyk/branch-A-Assertions

Add assertions to main code

shawn-nyk committed:
Merge branch 'master' into branch-A-CodeQuality

Sync branch-A-CodeQuality with master branch.

shawn-nyk committed:
Merge pull request #2 from shawn-nyk/branch-A-CodeQuality

Improve code quality

shawn-nyk committed:
Implement more flexible command syntax

There is only 1 valid command keyword per command and it is
case-sensitive.

Users will have to enter the full word and in the correct casing in
order for it to be recognised as a valid command keyword. This is an
unnecessary restriction.

Let's add a list of valid command keywords to each command such that
users can invoke a command via a number of different sensible keywords.
Also, let's make command keywords case-insensitive.

In this way, users have more flexibility in invoking commands.

shawn-nyk committed:
Merge branch 'branch-C-FriendlierSyntax'

Implement more flexible command syntax for the master branch version of
Duke.

shawn-nyk committed:
Refactor ui components into ui package

shawn-nyk committed:
Improve the GUI

Make the following GUI improvements:
* Increase width of application window and components to decrease
  number of times messages get wrapped
* Change the color of the text field and send button
* Change the font used throughout the application

shawn-nyk committed:
Fix JUnit tests

shawn-nyk committed:
Resize GUI

shawn-nyk committed:
Add a user guide

shawn-nyk committed:
Set theme jekyll-theme-leap-day

shawn-nyk committed:
Set theme jekyll-theme-merlot

shawn-nyk committed:
Set theme jekyll-theme-cayman

shawn-nyk committed:
Add more Javadocs

Add Javadocs to all public members of classes in main code.

shawn-nyk committed:
Add more JUnit tests

Add JUnit tests for the following classes:
* Response
* TaskList

shawn-nyk committed:
Merge branch 'master' of https://github.com/shawn-nyk/ip
